### ISSUE TITLE
UX: Cache "Your Repositories" list and allow refresh

### ISSUE THREAD

**Issue by joeljfischer** [ForkIssues/Tracker#1718]
When I'm setting up a new computer or, for some other reason, cloning a bunch of my projects, the repository list is extensive and can take a bit to load each time. Instead, the list should be cached (within reasonable limits, maybe even as low as 15 minutes?). There should also be a refresh button to force a refresh.

**Comment by DanPristupov** [comment]
> the repository list is extensive and can take a bit to load each time

Do you mean the repository list of GitHub? On my computer It takes about 1s to make request to the GitHub API.

**Comment by joeljfischer** [comment]
Yes, the GitHub list in the clone list. I'm a part of several orgs and a hundred or so repos are in the list. It seems to take 5-10 seconds to load each time. It's fine, but kind of annoying when I'm trying to clone 20 repos.
